### PR TITLE
Further limits on expensive path building

### PR DESCRIPTION
**Pull request by ctz**
This is intended to be complementary to #160 and addresses briansmith/webpki#276 in the same manner as NSS libmozpkix.

### CODE REVIEW

**Review by cpu — APPROVED** [review]
This looks great, thank you for tidying up some of the surrounding code. I peeked at the diff coverage dip and it's specific to the error ranking changes (which are much appreciated).

File: src/verify_cert.rs [hunk 1]
```diff
@@ -151,8 +151,7 @@ fn build_chain_inner(
             const MAX_SUB_CA_COUNT: usize = 6;
 
             if sub_ca_count >= MAX_SUB_CA_COUNT {
-                // TODO(XXX): Candidate for a more specific error - Error::PathTooDeep?
-                return Err(Error::UnknownIssuer);
+                return Err(Error::MaximumPathDepthExceeded);
```
Comment (stanal) [inline src/verify_cert.rs:154]: does this limit the length of certificate chain to 6?

Comment (ctz) [reply src/verify_cert.rs:154]: yes, though this error is not fatal so the search can continue to find valid shorter chains.

Comment (cpu) [reply src/verify_cert.rs:154]: Also, this branch only changes the error variant to be more specific and adds test coverage for this case. The limit being imposed is unchanged and inherited from briansmith/webpki.

Comment (cpu) [reply src/verify_cert.rs:154]: @stanal Do you have examples of real world certificate paths issued by a trust anchor participating in the web PKI that exceed this depth?

Comment (stanal) [reply src/verify_cert.rs:154]: > @stanal Do you have examples of real world certificate paths issued by a trust anchor participating in the web PKI that exceed this depth?
no, acturally , I cann't konw.  Although it does not exist for now, not means it won't exist in the future, may be will has one 7 or 8 ... and so on. I'm just not sure of that.

Comment (ctz) [reply src/verify_cert.rs:154]: I believe the limit of 6 comes from NSS libmozpkix -- eg https://github.com/nss-dev/nss/blob/master/lib/mozpkix/lib/pkixbuild.cpp#L322-L332 -- and therefore provides a bit of reassurance that people don't rely on certificate chains longer than this in the public internet.

Comment (stanal) [reply src/verify_cert.rs:154]: thanks! got it !

Comment (cpu) [reply src/verify_cert.rs:154]: As another data-point, while looking into something unrelated I noticed that Java+Android's [`PKIXBuilderParameters`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/PKIXBuilderParameters.html) defaults to a maximum path length of 5.

**Pull request merged**